test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -64,8 +61,41 @@ using ::grpc_core::experimental::StaticCrlProvider;
 namespace grpc_core {
 namespace testing {
 
+absl::StatusOr<std::shared_ptr<CrlProvider>>
+CreateDirectoryReloaderCrlProviderForTest(
+    absl::string_view directory, std::chrono::seconds refresh_duration,
+    std::function<void(absl::Status)> reload_error_callback,
+    std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine,
+    std::shared_ptr<Directory> directory_impl) {
+  if (directory_impl == nullptr) {
+    directory_impl = std::make_shared<Directory>(directory);
+  }
+  if (event_engine == nullptr) {
```

Comment:
I think every test should use the fuzzing EE, so that we don't wind up executing any timer callbacks if the test doesn't care about them.

I suggest building a framework to avoid too much boilerplate in the individual tests:

```
class DirectoryReaderCrlProviderTest : public ::testing::Test {
 protected:
  // Tests that want a fake directory reader can call this without setting the last parameter.
  absl::StatusOr<std::shared_ptr<CrlProvider>> CreateCrlProvider(
      std::chrono::seconds refresh_duration,
      std::function<void(absl::Status)> reload_error_callback,
      std::shared_ptr<DirectoryReader> directory_reader = nullptr) {
    if (directory_reader == nullptr) directory_reader = directory_reader_;
    auto provider = std::make_shared<experimental::DirectoryReloaderCrlProvider>(
        refresh_duration, std::move(reload_error_callback), event_engine_,
        std::move(directory_reader));
    absl::Status initial_status = provider->Update();
    if (!initial_status.ok()) return initial_status;
    provider->ScheduleReload();
    return provider;
  }

  // Tests that want a real directory can call this instead of the above.
  absl::StatusOr<std::shared_ptr<CrlProvider>> CreateCrlProvider(
      absl::string_view directory, std::chrono::seconds refresh_duration,
      std::function<void(absl::Status)> reload_error_callback) {
    return CreateCrlProvider(
        refresh_duration, std::move(reload_error_callback),
        std::make_shared<DirectoryReader>(directory));
  }

  auto directory_reader_ = std::make_shared<FakeDirectoryReader>();
  auto event_engine_ =
      std::make_shared<grpc_event_engine::experimental::FuzzingEventEngine>(
          grpc_event_engine::experimental::FuzzingEventEngine::Options(),
          fuzzing_event_engine::Actions());
};
```

Now the individual tests just need to call the appropriate variant of `CreateCrlProvider()`, and they get a provider with the right dependencies injected.

Reply:
Made the change, good call - this is much cleaner